app/forms/network-interface-create.tsx

```diff
@@ -80,7 +80,16 @@ export default function CreateNetworkInterfaceForm({
         required
         control={form.control}
       />
-      <TextField name="ip" label="IP Address" control={form.control} />
+      <TextField
+        name="ip"
+        label="IP Address"
+        control={form.control}
+        onBlur={() => {
+          if (form.getValues('ip')?.trim() === '') {
+            form.setValue('ip', undefined)
+          }
+        }}
```

Comment:
Say you have this field focused and you click submit. That means you have blur and submit happening at roughly the same time. Is there a race condition there? Is it possible submit could run before this gets a chance to fix the value? If so, it might be necessary to intercept `onChange` somehow instead, maybe with a `transform` prop?

See for example: https://github.com/orgs/react-hook-form/discussions/8090#discussioncomment-2430118

After writing the above, I now see that's exactly what Justin is doing in #1812, so maybe I'm just remembering that.

https://github.com/oxidecomputer/console/pull/1812/files#diff-6333dd86e85b981135bbc9c2392fbd9f9d058b9ac1ea86159dd8275e9e50b404R94

This also says it's better to hook into onChange: https://react-hook-form.com/advanced-usage#TransformandParse